src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
If I understand correctly, this change was made in order to fix the tests. I don't think it's the right fix.

We wanted to test that JSONified dates would be parsed correctly. In order to test that, we relied on the fact that `$interpolate` would convert date objects to JSON. Now that `$interpolate` does not convert them to JSON (but calls `toString()` instead), we should change the _tests_, so that JSONified dates are still passed to `max`/`min`.

With the current fix, we have a test that is supposed to test parsing JSONified dates, but it tests stringified dates instead, right ?

--
Part of the big BC will be that `max`/`min` won't be able to handle interpolated date objects. One will need to either use `ngMax/Min` or change `{{ myDateObj }}` to `{{ myDateObj.toISOString() }}` or `{{ myDateObj | json }}` (or manually JSONify the date object in the controller).
I don't think it's too bad, though.

Another idea would be to check if `new Date(iso).toString() === iso`. If this is true, then `iso` must be the `toString()` output of a Date equal to `new Date(iso)`.

--
BTW, `Date.parse()` returns the number of milliseconds. Don't we need a Date object to be returned ?